kin-devplatform-sdk/src/main/java/kin/devplatform/KinEcosystemInitiator.java

```diff
@@ -76,85 +80,109 @@ public void internalInit(Context context) {
 	/**
 	 * Uses for external (public API) initialization and jwt login.
 	 */
-	public void externalInit(Context context, KinEnvironment environment, @NonNull SignInData signInData,
-		final KinCallback<Void> loginCallback) {
+	public void externalInit(Context context, String appId, KinEnvironment environment, @NonNull SignInData signInData,
```

Comment:
you can parse the body of the JWT, also check if any of your partners using the `WhiteList` option, i suggest to remove the `WhiteList` and just use jwt and parse the body to get the relevant data you need such as appId.

Reply:
which dependency? the developer also specifies the appId on the jwt he doesn't need to specify it twice.
To parse JWT is simple Base64